sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableServiceImpl.java

```diff
@@ -210,6 +234,246 @@ public Row getCurrentRow() throws NoSuchElementException {
     }
   }
 
+  @VisibleForTesting
+  static class BigtableMiniBatchReaderImpl implements Reader {
+    private BigtableSession session;
+    private final BigtableSource source;
+    private Row currentRow;
+    private Queue<FlatRow> buffer;
+    private RowSet rowSet;
+    private ServiceCallMetric serviceCallMetric;
+    private Future<List<FlatRow>> future;
+    private ByteString lastFetchedRow;
+    private boolean lastFillComplete;
+
+    private int miniBatchLimit = DEFAULT_MINI_BATCH_SIZE;
+    private long bufferSizeLimit = DEFAULT_BYTE_LIMIT;
+    private int miniBatchWaterMark;
```

Comment:
please add final modifiers for all properties that are const

Reply:
Done.